src/transport/transport.interface.ts

```diff
@@ -1,10 +1,19 @@
 import { createInterface } from 'common/di/functions/createInterface';
+import { MessageType } from '../messages';
 
 export const TransportInterface =
   createInterface<TransportInterface>('TransportInterface');
 
 export interface TransportInterface {
-  publish<T>(topic: string, message: T): Promise<void>;
-  subscribe<T>(topic: string, cb: (message: T) => Promise<void>): Promise<void>;
+  publish<T>(
+    topic: string,
+    message: T,
+    messageType: MessageType,
+  ): Promise<void>;
+  subscribe<T>(
+    topic: string,
+    messageType: string,
```

Comment:
`messageType` should have `MessageType` type.

Reply:
Fixed